app/uk/gov/hmrc/softdrinksindustrylevystub/services/SubscriptionGenerator.scala

```diff
@@ -16,167 +16,147 @@
-    Gen.alphaNumStr.sometimes |@|       // siteReference
-    Gen.date(2017, 2020).sometimes |@|  // dateOfClosure
-    Gen.alphaNumStr.sometimes |@|       // siteClosureReason
-    Gen.alphaNumStr.sometimes |@|       // tradingName
-    Gen.alphaNumStr.sometimes |@|       // newSiteReference
-    addressGen |@|                      // address
-    siteTypeGen.sometimes               // typeOfSite
+    Gen.const("1")                                           |@| // action
+    Gen.alphaNumStr                                          |@| // tradingName
+    Gen.alphaNumStr                                          |@| // newSiteReference
+    businessContactGen                                       |@| // address
+    Gen.oneOf("1","2")                                           // typeOfSite
   }.map(Site.apply)
 
   private lazy val addressGen: Gen[Address] = {
-    Gen.boolean |@|                     // addressNotInUk
-    Gen.alphaNumStr |@|                 // addressLine1
-    Gen.alphaNumStr |@|                 // addressLine2
-    Gen.alphaNumStr.sometimes |@|       // addressLine3
-    Gen.alphaNumStr.sometimes |@|       // addressLine4
-    Gen.alphaNumStr |@|                 // postcode
-    Gen.alphaNumStr.sometimes |@|       // nonUkCountry
-    Gen.alphaNumStr |@|                 // telephoneNumber
-    Gen.alphaNumStr.sometimes |@|       // mobileNumber
-    Gen.alphaNumStr |@|                 // emailAddress
-    Gen.alphaNumStr.sometimes           // faxNumber
+    Gen.boolean                                              |@| // notUKAddress
+    Gen.alphaNumStr                                          |@| // line1
+    Gen.alphaNumStr                                          |@| // line2
+    Gen.alphaNumStr.sometimes                                |@| // line3
+    Gen.alphaNumStr.sometimes                                |@| // line4
+    Gen.alphaNumStr.sometimes                                |@| // postCode
+    pattern"AA".gen.sometimes                                    // country
   }.map(Address.apply)
 
-  private lazy val bankDetailsGen: Gen[BankDetails] = {
-    Gen.boolean |@|                     // directDebit
-    Gen.alphaNumStr.sometimes |@|       // accountName
-    Gen.alphaNumStr.sometimes |@|       // accountNumber
-    Gen.alphaNumStr.sometimes |@|       // sortCode
-    Gen.alphaNumStr.sometimes           // buildingSocietyRollNumber
-  }.map(BankDetails.apply)
-
   private lazy val litresProducedGen: Gen[LitresProduced] = {
-    Gen.choose(1, 1000000).sometimes |@|  // producedLower
-    Gen.choose(1, 1000000).sometimes |@|  // producedHigher
-    Gen.choose(1, 1000000).sometimes |@|  // importedLower
-    Gen.choose(1, 1000000).sometimes |@|  // importedHigher
-    Gen.choose(1, 1000000).sometimes |@|  // packagedLower
-    Gen.choose(1, 1000000).sometimes      // packagedHigher
+    Gen.choose(1, 9999999999999L).sometimes                  |@| // litresProducedUKHigher
```

Comment:
Should we externalise this to a val as it is used a lot?

Reply:
Done.